helix-core/src/main/java/org/apache/helix/ShardingKeyTrieNode.java

```diff
@@ -0,0 +1,66 @@
+package org.apache.helix;
```

Comment:
Please create a separate package for this instead of putting it in `org.apache.helix`. Since this will be used by helix-rest, I am fine with creating a package called "metadata-store" or something like that in helix-rest for the time being.